kafka-connect/src/main/java/io/tabular/iceberg/connect/IcebergSinkTask.java

```diff
@@ -75,114 +50,57 @@ public void start(Map<String, String> props) {
+        } catch (Exception e) {
+          LOG.warn(
+              "An error occurred while trying to close {} instance, ignoring...",
+              closeable.getClass().getSimpleName(),
+              e);
+        }
       }
     }
-
-    LOG.info("Starting commit worker");
-    IcebergWriterFactory writerFactory = new IcebergWriterFactory(catalog, config);
-    worker = new Worker(config, clientFactory, writerFactory, context);
-    worker.syncCommitOffsets();
-    worker.start();
   }
 
-  @VisibleForTesting
-  boolean isLeader(Collection<MemberDescription> members, Collection<TopicPartition> partitions) {
-    // there should only be one task assigned partition 0 of the first topic,
-    // so elect that one the leader
-    TopicPartition firstTopicPartition =
-        members.stream()
-            .flatMap(member -> member.assignment().topicPartitions().stream())
-            .min(new TopicPartitionComparator())
-            .orElseThrow(
-                () -> new ConnectException("No partitions assigned, cannot determine leader"));
+  private void clearState() {
+    close(writer);
+    writer = null;
 
-    return partitions.contains(firstTopicPartition);
+    close(committer);
+    committer = null;
   }
 
   @Override
-  public void close(Collection<TopicPartition> partitions) {
-    close();
-  }
-
-  private void close() {
-    clearObjectState();
-  }
-
-  private void clearObjectState() {
-    if (worker != null) {
-      worker.stop();
-      worker = null;
-    }
-
-    if (coordinatorThread != null) {
-      coordinatorThread.terminate();
-      coordinatorThread = null;
-    }
+  public void open(Collection<TopicPartition> partitions) {
+    // destroy any state if KC re-uses object
+    clearState();
 
-    if (catalog != null) {
-      if (catalog instanceof AutoCloseable) {
-        try {
-          ((AutoCloseable) catalog).close();
-        } catch (Exception e) {
-          LOG.warn("An error occurred closing catalog instance, ignoring...", e);
-        }
-      }
-      catalog = null;
-    }
+    this.writer = new WriterImpl(context, config);
+    this.committer = new CommitterImpl(context, config);
   }
 
   @Override
   public void put(Collection<SinkRecord> sinkRecords) {
-    if (sinkRecords != null && !sinkRecords.isEmpty() && worker != null) {
-      worker.save(sinkRecords);
-    }
-    processControlEvents();
+    writer.put(sinkRecords);
+    committer.commit(writer);
   }
 
   @Override
-  public void flush(Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
-    processControlEvents();
-  }
-
-  private void processControlEvents() {
-    if (coordinatorThread != null && coordinatorThread.isTerminated()) {
-      throw new NotRunningException("Coordinator unexpectedly terminated");
-    }
-    if (worker != null) {
-      worker.process();
-    }
+  public Map<TopicPartition, OffsetAndMetadata> preCommit(
+      Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
+    return ImmutableMap.of();
```

Comment:
Not committing to the connect consumer group via this method anymore, all offsets are managed manually.